include/unifex/let_value_with_stop_token.hpp

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <unifex/config.hpp>
+#include <unifex/just.hpp>
+#include <unifex/let_value.hpp>
+#include <unifex/receiver_concepts.hpp>
+#include <unifex/sender_concepts.hpp>
+#include <unifex/inplace_stop_token.hpp>
+#include <unifex/execution_policy.hpp>
+#include <unifex/type_list.hpp>
+
+#include <unifex/detail/prologue.hpp>
+
+namespace unifex {
+
+namespace _let_v_w_stop_tok {
+
+template<typename Operation, typename Receiver>
+struct _stop_token_receiver {
+    class type;
+};
+
+template<typename InnerOp, typename Receiver>
+struct _stop_token_operation {
+  struct type;
+};
+
+template <typename InnerOp, typename Receiver>
+using operation =  typename _stop_token_operation<InnerOp, remove_cvref_t<Receiver>>::type;
+
+template<typename Operation, typename Receiver>
+using stop_token_receiver = typename _stop_token_receiver<Operation, Receiver>::type;
+
+template<typename Operation, typename Receiver>
+class _stop_token_receiver<Operation, Receiver>::type {
+public:
+    explicit type(Operation& op, Receiver&& r) noexcept(std::is_nothrow_move_constructible_v<Receiver>) :
+        op_(&op), receiver_{(Receiver &&)r}
```

Comment:
The formatting is off here,

```suggestion
    explicit type(Operation& op, Receiver&& r) noexcept(std::is_nothrow_move_constructible_v<Receiver>)
        : op_(&op), receiver_{(Receiver &&)r}
```